opm/common/utility/OpmInputError.hpp

```diff
@@ -0,0 +1,91 @@
+/*
+  Copyright 2020 Equinor ASA.
+
+  This file is part of the Open Porous Media project (OPM).
+
+  OPM is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  OPM is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with OPM.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include <stdexcept>
+#include <string>
+
+#include <fmt/format.h>
+
+#ifndef OPM_ERROR_HPP
+#define OPM_ERROR_HPP
+
+namespace Opm {
+
+/*
+  The OpmInputError is a custom exception class which can be used to signal
+  errors in input handling. The importance of the OpmInputError exception is
+  *not* the tecnical functionality it provides, but rather the convention
+  surrounding it, when and how it should be used.
+
+  The OpmInputError should be used in situations which are "close to user
+  input", the root cause can either be incorrect user input or a bug/limitation
+  in opm. OpmInputError should only be used in situations where we have a good
+  understanding of the underlying issue, and can provide a good error message.
+
+  The local error handling should be complete when the OpmInputError is
+  instantiated, it should not be caught and rethrown in order to e.g. add
+  additional context or log messages. In order to avoid inadvertendly catching
+  this exception in a catch all block.
+*/
+
+
+
+class OpmInputError : public std::exception {
+public:
+    /*
+      The message string can optionally contain {} markers which will be
+      replaced with the fields from location argument:
+
+        {keyword} -> loc.keyword
+        {file} -> loc.filename
+        {name} -> loc.lineno
+
+      So OpmInputError("Error at line {2} while parsing keyword {0} in file {1}", KeywordLocation("EQUIL", "equil.inc", 100))
```

Comment:
Above there are named arguments while here there are numbered. I think your example still works, but perhaps make it clearer that you can use either?